qiskit/qobj/converters/pulse_instruction.py

```diff
@@ -322,180 +316,327 @@ def convert_bundled_acquires(
 
         return self._qobj_model(**command_dict)
 
-    @bind_instruction(instructions.SetFrequency)
-    def convert_set_frequency(self, shift, instruction):
-        """Return converted `SetFrequencyInstruction`.
+    def _convert_SetFrequency(
+        self,
+        instruction: instructions.SetFrequency,
+        time_offset: int,
+    ) -> PulseQobjInstruction:
+        """Return converted `SetFrequency`.
 
         Args:
-            shift (int): Offset time.
-            instruction (SetFrequency): set frequency instruction.
+            instruction: Qiskit Pulse set frequency instruction.
+            time_offset: Offset time.
 
         Returns:
-            dict: Dictionary of required parameters.
+            Qobj instruction data.
         """
         command_dict = {
             "name": "setf",
-            "t0": shift + instruction.start_time,
+            "t0": time_offset + instruction.start_time,
             "ch": instruction.channel.name,
             "frequency": instruction.frequency / 1e9,
         }
         return self._qobj_model(**command_dict)
 
-    @bind_instruction(instructions.ShiftFrequency)
-    def convert_shift_frequency(self, shift, instruction):
+    def _convert_ShiftFrequency(
+        self,
+        instruction: instructions.ShiftFrequency,
+        time_offset: int,
+    ) -> PulseQobjInstruction:
         """Return converted `ShiftFrequency`.
 
         Args:
-            shift (int): Offset time.
-            instruction (ShiftFrequency): Shift frequency instruction.
+            instruction: Qiskit Pulse shift frequency instruction.
+            time_offset: Offset time.
 
         Returns:
-            dict: Dictionary of required parameters.
+            Qobj instruction data.
         """
         command_dict = {
             "name": "shiftf",
-            "t0": shift + instruction.start_time,
+            "t0": time_offset + instruction.start_time,
             "ch": instruction.channel.name,
             "frequency": instruction.frequency / 1e9,
         }
         return self._qobj_model(**command_dict)
 
-    @bind_instruction(instructions.SetPhase)
-    def convert_set_phase(self, shift, instruction):
+    def _convert_SetPhase(
```

Comment:
I'm surprised pylint doesn't complain about this being camel case.

Reply:
Yes, this was what I wanted to change. Now we can use method dispatch in terra, I updated the implementation of these method in [f1ad415](https://github.com/Qiskit/qiskit-terra/pull/8839/commits/f1ad41575d63c1b99f1f81b5276125fe4d347c6c)